src/Agent.Sdk/SecretMasking/LoggedSecretMasker.cs

```diff
@@ -1,16 +1,23 @@
-using Microsoft.TeamFoundation.DistributedTask.Logging;
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
 using System;
+using ValueEncoder = Microsoft.TeamFoundation.DistributedTask.Logging.ValueEncoder;
+using ISecretMaskerVSO = Microsoft.TeamFoundation.DistributedTask.Logging.ISecretMasker;
+using System.Text.RegularExpressions;
 
-namespace Agent.Sdk.Util
+namespace Agent.Sdk.SecretMasking
 {
     /// <summary>
     /// Extended secret masker service, that allows to log origins of secrets
     /// </summary>
     public class LoggedSecretMasker : ILoggedSecretMasker
     {
+        public bool UsePerformanceEnhancements { get; set; } = false;
```

Comment:
minor: For correct semantics, UsePerformanceEnhancements should be in the constructor or AddRegex as a paramter.  (e.g. should not be mutable by the caller, should be supplied prior to AddRegex call)

Reply:
@merlynomsft I removed this one, and use new approach by default in new SecretMasker